### PR TITLE
Refactor ProjectValidationResult*ElParam classes

### LINKED ISSUE #343: Refactor ProjectValidationResult*ElParam classes
When using validations, I found out that `ProjectValidationResultGdcTimeElParam`'s `vals` field is always null. This means that this field is not necessary and should be removed. 

Another thing is that list fields in `ProjectValidationResultSliElParam` class (`ids`, `vals`) represent tuples with values on the same index in these two lists. For better comparison is to add these values into some collection of tuples.

### PR DESCRIPTION
**Pull request by peter-plochan**
Removed useless field `vals` from ProjectValidationResultGdcTimeElParam
Removed ProjectValidationResultElParam
Add `asMap()` method generating collection of tuples in ProjectValidationResultSliElParam

Fixes #343

### CODE REVIEW

File: src/main/java/com/gooddata/project/ProjectValidationResultGdcTimeElParam.java [hunk 1]
```diff
@@ -6,31 +6,52 @@
...
+        return ids;
+    }
 
-        ProjectValidationResultGdcTimeElParam that = (ProjectValidationResultGdcTimeElParam) o;
+    /**
+     * @deprecated for backward compatibility only. Do not use this method, it always returns null.
+     */
+    @Deprecated
+    public List<String> getVals() {
+        return null;
```
Comment (jimirocks) [inline src/main/java/com/gooddata/project/ProjectValidationResultGdcTimeElParam.java]: - [ ] rather return empty list

File: src/main/java/com/gooddata/project/ProjectValidationResultSliElParam.java [hunk 2]
```diff
@@ -1,25 +1,65 @@
...
 
+    public List<String> getIds() {
+        return ids;
+    }
+
+    public List<String> getVals() {
+        return vals;
+    }
+
+    public Map<String, String> asMap() {
```
Comment (jimirocks) [inline src/main/java/com/gooddata/project/ProjectValidationResultSliElParam.java]: - [ ] add some javadoc

File: src/main/java/com/gooddata/project/ProjectValidationResultSliElParam.java [hunk 3]
```diff
@@ -1,25 +1,65 @@
...
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 @JsonTypeName("sli_el")
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class ProjectValidationResultSliElParam extends ProjectValidationResultElParam {
+public class ProjectValidationResultSliElParam extends ProjectValidationResultParam {
+
+    /**
```
Comment (jimirocks) [inline src/main/java/com/gooddata/project/ProjectValidationResultSliElParam.java]: - [ ] please move this javadoc to getters of ids and vals

File: src/main/java/com/gooddata/project/ProjectValidationResultGdcTimeElParam.java (hunk 1 above)
Comment (peter-plochan) [inline src/main/java/com/gooddata/project/ProjectValidationResultGdcTimeElParam.java]: The method returned `null` value before the fix (JSON structure doesn't contain `vals` node) so I would prefer to preserve this as it is for ensuring previous behaviour.

**Pull request merged**